### PR TITLE
Ensure Tesla is loaded before defining MetaLogger middleware

### PR DESCRIPTION
**Pull request by nvieirafelipe**
Closes: #5

### CODE REVIEW

**Review by sergioaugrod — APPROVED** [review]

File: lib/tesla/middleware/meta_logger.ex [hunk 1]
```diff
@@ -1,134 +1,136 @@
-defmodule Tesla.Middleware.MetaLogger do
-  @moduledoc """
-  Tesla middleware to log requests and responses.
+if Code.ensure_loaded?(Tesla) do
```
Comment (zekus) [inline lib/tesla/middleware/meta_logger.ex:1]: 😱 this is super ugly. is there no other way? 
if not, then we should extract this module into another library and add it explicitly where we want to use it.

Comment (prodis) [reply lib/tesla/middleware/meta_logger.ex:1]: Maybe a wrapper function in `Tesla.run/2`.

Comment (nvieirafelipe) [reply lib/tesla/middleware/meta_logger.ex:1]: I know but this is exactly what we need. We have Tesla as optional dependency and this module depends on it so when the dependency is not there we shouldn't compile it.

Comment (prodis) [reply lib/tesla/middleware/meta_logger.ex:1]: I changed my mind after the discussion in Slack: https://findhotel.slack.com/archives/GJDF4TX38/p1619107278055000. 

It resolves at compile time, much better than runtime.

**Review by prodis — APPROVED** [review]

**Pull request merged**